admin/release.py

```diff
@@ -448,16 +470,89 @@ def upload_rpms(scratch_directory, target_bucket, version, build_server):
         )
 
 
-def publish_rpms_main(args, base_path, top_level):
+def create_pip_index(scratch_directory, packages):
     """
-    The ClusterHQ yum repository contains packages for Flocker, as well as the
-    dependencies which aren't available in Fedora 20 or CentOS 7. It is
-    currently hosted on Amazon S3. When doing a release, we want to add the
-    new Flocker packages, while preserving the existing packages in the
-    repository. To do this, we download the current repository, add the new
-    package, update the metadata, and then upload the repository.
+    Create an index file for pip.
 
-    :param list args: The arguments passed to the script.
+    :param FilePath scratch_directory: Temporary directory to create index in.
+    :param list packages: List of bytes, filenames of packages to be in the
+        index.
+    """
+    index_file = scratch_directory.child('index')
+    with index_file.open('w') as f:
+        f.write('<!--This is an index for pip-->\n')
+        for package in packages:
+            if package.endswith('.whl'):
+                f.write('<a href={destination}>{title}</a><br/>\n'.format(
+                    destination=quoteattr(package),
+                    title=escape(package)))
+    return index_file
+
+
+@do
+def upload_pip_index(scratch_directory, target_bucket):
+    """
+    Upload an index file for pip to S3.
+
+    :param FilePath scratch_directory: Temporary directory to create index in.
+    :param bytes target_bucket: S3 bucket to upload index to.
+    """
+    packages = yield Effect(
+        ListS3Keys(bucket=target_bucket,
+                   prefix='python/'))
+
+    index_path = create_pip_index(
+        scratch_directory=scratch_directory,
+        packages=packages)
+
+    yield Effect(
+        UploadToS3(
```

Comment:
We need to ensure that the content type is `text/html`, otherwise pip will ignore it.